docs/key_modules/workforce.md

```diff
@@ -0,0 +1,255 @@
+# Workforce
+
+> *Workforce is a system where multiple agents/role-playing systems work together to solve tasks. By using Workforce, users can quickly set up a multi-agent task solving system with customized configurations. In this section, we will give a brief view on the architecture of workforce, and how you can configure and utilize it to solve problems.*
+
+## Architecture
+
+Workforce follows a tree-shaped architecture, with one or more managers (internal nodes) managing multiple fine-tuned workers with different roles (leaf nodes). A manager is responsible of managing and coordinating its child nodes and can also be a child of another manager; a worker, on the other hand, must be a child node of a manager.
+
+> For now, CAMEL only supports the mode with one manager as root and multiple workers as direct leaf nodes of the root.
+
+All manager and worker nodes share a single channel, where they can post and receive tasks, broadcast task results, etc.
+
+## Workflow
+
+In this section we are going to talk about how workforce works under the hood, briefly.
+
+As mentioned above, workforce is a system to solve tasks. Based on this, intrinsically, we make the initial input to a workforce as be a `Task` instance describing the main task that should be solved. This task will then be passed into the root of the workforce as the initial task of it.
+
+After starting up, the first thing that the root manager will do is decomposing the main task into multiple subtasks, so that it can be solved step by step.
+
+Afterwards, the manager will assign each of them to a worker that is most capable of doing this task, according to the global information it has, such as the description of each worker it manages. The subtasks are not posted all in a time. Instead, due to the dependency relationships, it will only post tasks with all the dependencies (also tasks) completed.
+
+Each worker will take the assigned tasks and try to complete it. If the task is successfully done, the worker will update the result of the task and notify the manager, so that the manager can post new tasks that depends on it.
+
+However, task can fail. We don’t want to see this happen, but things go wrong. Fortunately, managers can fix this. When a task fails, the worker will report the failure to the manager, and manager can take one of the following actions:
+
+1. decompose the task into smaller ones and assign them again
+2. create a new worker that is capable of doing the task
+
+For now, if the task has already been decomposed a lot of times, the manager will take the new worker creation action; if not, the manager will simply take the decomposition action.
+
+When all the subtasks are finished, their results will be composed into a single, comprehensive result and then get updated into the main task instance and returned to the user.
+
+## A Full Demo of Workforce
+
+Below is a comprehensive example of the using of workforce.
```

Comment:
IMO we can remove this part as it's a little bit redundant, WDT?

Reply:
You mean we just keep the step-by-step instructions?